pkg/indexers/indexers.go

```diff
@@ -124,3 +125,18 @@ func ByIndex[T runtime.Object](indexer cache.Indexer, indexName, indexValue stri
 
 	return ret, nil
 }
+
+// ByPathAndName returns the instance of T from the indexer with the matching path and name.
```

Comment:
mention that this is partial and depends on the path annotation.

Also: it matches both a canonical path **and** the cluster name.

Reply:
Full path or any permutation?

Reply:
Updated the comment; PTAL